File: DynamicProgramming1/FibonacciFunction/CFibonacciFunction/FibonacciFunction.c
#include <stdio.h>

#define MAX 41

int memo_0[MAX] = {1, 0};
int memo_1[MAX] = {0, 1};

void fibonacci_function()
{
	for (int i = 2; i < MAX; i++)
	{
		memo_0[i] = memo_0[i - 1] + memo_0[i - 2];
		memo_1[i] = memo_1[i - 1] + memo_1[i - 2];
	}
}

int main()
{
	int t, n;

	fibonacci_function();

	scanf_s("%d", &t);

	for (int i = 0; i < t; i++)
	{
		scanf_s("%d", &n);
		printf("%d %d\n", memo_0[n], memo_1[n]);
	}

	return 0;
}

File: DynamicProgramming1/FibonacciFunction/GoFibonacciFunction/FibonacciFunction.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const (
	max = 41
)

var (
	memo0 [max]int
	memo1 [max]int
)

func fibonacciFunction() {
	memo0[0] = 1
	memo1[0] = 0
	memo0[1] = 0
	memo1[1] = 1

	for i := 2; i < max; i++ {
		memo0[i] = memo0[i-1] + memo0[i-2]
		memo1[i] = memo1[i-1] + memo1[i-2]
	}
}

func main() {
	br := bufio.NewReader(os.Stdin)
	bw := bufio.NewWriter(os.Stdout)

	fibonacciFunction()

	t := 0
	fmt.Fscanln(br, &t)

	for i := 0; i < t; i++ {
		n := 0
		fmt.Fscanln(br, &n)
		fmt.Fprintf(bw, "%d %d\n", memo0[n], memo1[n])
	}

	bw.Flush()
}
